feat(tab): adds closable attribute and close event (VIV-2030)

TaylorJ76 opened the pull request.

RichardHelm commented on libs/components/src/lib/tab/tab.template.ts:
```diff
@@ -34,15 +50,20 @@ const getClasses = ({
  */
 export function TabTemplate<T extends Tab>(context: ElementDefinitionContext) {
 	const affixIconTemplate = affixIconTemplateFactory(context);
+	const buttonTag = context.tagFor(Button);
```
Button registries need to be added to `tabRegistries` in `definition.ts` so that Button will registered as well when Tab is registered

rachelbt commented on libs/components/src/lib/tab/tab.template.ts:
```diff
@@ -23,9 +25,23 @@ const getClasses = ({
+function renderDismissButton(buttonTag: string) {
+	return html<Tab>`
+		<${buttonTag}
+			aria-label=""
```
also - the tab now is announced as group and referring the button as - one more item but no way to know what it is :(

rachelbt reviewed (changes requested):
didn't went over all the code yet, tested keyboad navigation and screen-reader and its not 100%.
When tab is removed (also - there's no way to know it is removable), the focus disappears, shouldn't it go the the latest tab?
Also - if all tabs are deleted it creates a strange use case:
![image](https://github.com/user-attachments/assets/ddbda6ec-3d1f-4f88-bca3-33b34b3a94c9)
maybe we need to add that if only one is left it can't be removed?

rachelbt commented on libs/components/src/lib/tab/tab.scss:
```diff
@@ -120,3 +128,10 @@ slot[name='icon'] {
+	position: absolute;
+	inset-block-start: 50%;
+	inset-inline-end: 4px;
+	margin-block-start: -12px;
```
```suggestion
	transform: translateY(-50%);
```

TaylorJ76 commented:
> didn't went over all the code yet, tested keyboad navigation and screen-reader and its not 100%. When tab is removed (also - there's no way to know it is removable), the focus disappears, shouldn't it go the the latest tab? Also - if all tabs are deleted it creates a strange use case: ![image](https://private-user-images.githubusercontent.com/7512792/371025171-ddbda6ec-3d1f-4f88-bca3-33b34b3a94c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2NzkyNDUsIm5iZiI6MTcyNzY3ODk0NSwicGF0aCI6Ii83NTEyNzkyLzM3MTAyNTE3MS1kZGJkYTZlYy0zZDFmLTRmODgtYmNhMy0zM2IzNGIzYTk0YzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MzBUMDY0OTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGJlNWY1Yjc0NTIyOTRjZjU1MGMyZGVlZWUxNTM3NGUzZDEwOTllN2FlOWY2ZWQwYWZhYTBlZmRiMmJjMTM5MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.AApqtc72cjB0fbxFuVqe_7SBHSepfLzsuIenpaDcOdM) maybe we need to add that if only one is left it can't be removed?

As the removal of the tab is up to the consumer, I think handling this use case is also the consumers responsibility. I have update the examples so that when the last tab is removed, the whole tab component is also removed. And I have updated the warning note on the tab page.

rachelbt reviewed (commented):
👍 just ine comment about the icon

rachelbt reviewed (approved):
looks great
